### ISSUE TITLE
Table cell containing `|` fails to parse / lint correctly

### ISSUE THREAD

**Issue by Ir1d** [remarkjs/remark-lint#254]
### Subject of the issue

I was using remark-lint on the following md doc:

```
| Pascal      | C++        |     |     |
| ---------- | ---------- | --- | --- |
|  `not(a)`   |  `!a`      |     |     |
|  `a and b`  |  `a && b`  |     |     |
|  `a or b`   | `a||b` | | |
```

After Linting I get:

```
| Pascal      | C++        |     |     |     |     |
| ----------- | ---------- | --- | --- | --- | --- |
|  `not(a)`   |  `!a`      |     |     |     |     |
|  `a and b`  |  `a && b`  |     |     |     |     |
|  `a or b`   | \`a        |     | b\` |     |     |

```

In `--inspect` I noticed that:

```
   └─ tableRow[6] (5:1-5:29, 163-191)
      ├─ tableCell[1] (5:4-5:14, 166-176)
      │  └─ inlineCode: "a or b" (5:4-5:12, 166-174)
      ├─ tableCell[1] (5:17-5:19, 179-181)
      │  └─ text: "`a" (5:17-5:19, 179-181)
      ├─ tableCell[0] (5:20-5:20, 182-182)
      ├─ tableCell[1] (5:21-5:23, 183-185)
      │  └─ text: "b`" (5:21-5:23, 183-185)
      ├─ tableCell[0] (5:26-5:26, 188-188)
      └─ tableCell[0] (5:28-5:28, 190-190)
```

I wonder how to make my table support cells containing `|`? Thanks in advance

**Comment by wooorm** [comment]
Tables are a GFM (GitHub) feature. They render here the same as what remark sees:

| Pascal      | C++        |     |     |
| ---------- | ---------- | --- | --- |
|  `not(a)`   |  `!a`      |     |     |
|  `a and b`  |  `a && b`  |     |     |
|  `a or b`   | `a||b` | | |

You can escape pipes in code in table cells (note that you can’t escape any other characters in table cells):

```markdown
| Pascal      | C++        |     |     |
| ---------- | ---------- | --- | --- |
|  `not(a)`   |  `!a`      |     |     |
|  `a and b`  |  `a && b`  |     |     |
|  `a or b`   | `a\|\|b` | | |
```

| Pascal      | C++        |     |     |
| ---------- | ---------- | --- | --- |
|  `not(a)`   |  `!a`      |     |     |
|  `a and b`  |  `a && b`  |     |     |
|  `a or b`   | `a\|\|b` | | |
